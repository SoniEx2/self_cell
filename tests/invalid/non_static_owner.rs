use std::cell::Cell;

use once_self_cell::unsync::OnceSelfCell;

fn main() {
    let owner = String::from("bleib");
    let c: OnceSelfCell<&String, Cell<&String>> = OnceSelfCell::new(&owner, |_| panic!());
}
