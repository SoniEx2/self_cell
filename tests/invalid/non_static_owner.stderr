error[E0106]: missing lifetime specifier
  --> $DIR/non_static_owner.rs:11:40
   |
11 | sync_once_self_cell!(BorrowFromStack, &'_ String, Dependent<'_>,);
   |                                        ^^ expected named lifetime parameter
   |

error[E0621]: explicit lifetime required in the type of `owner`
  --> $DIR/non_static_owner.rs:11:1
   |
11 | sync_once_self_cell!(BorrowFromStack, &'_ String, Dependent<'_>,);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^----------^^^^^^^^^^^^^^^^^^
   | |                                     |
   | |                                     help: add explicit lifetime `'static` to the type of `owner`: `&'static String`
   | lifetime `'static` required
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
